lib/cloud_controller/deployment_updater/calculators/all_instances_routable.rb

```diff
@@ -0,0 +1,26 @@
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Calculators
+      class AllInstancesRoutable
+        attr_reader :deployment, :logger
+
+        def initialize(deployment, logger)
+          @deployment = deployment
+          @logger = logger
+        end
+
+        def call
```

Comment:
**suggestion** I'm not sure about having `call` be the general purpose function call for all the classes here, especially when they return different thing (bool/ints/processes). Are you envisioning some future refactor benefit?

Reply:
I think the main benefit here is to prevent mental overhead of trying to remember which Calculator has which specific method, when they are all designed to do one thing and one thing only. It's simpler (and less redundant feeling) than calling something like `AllInstancesRoutable.new(...).all_instances_routable?`

That said, I'm open to a different name/pattern.

Historically, `call` is usually chosen for these kinds of single-purpose Ruby objects as it allows them to be used interchangeably with lambdas, as they are executed with `call`.

Reply:
Makes sense, then I if its`call` is basically determined by whether the refactor is single funciton classes or not.